Spring Config: Using `@ConfigurationProperties` to inject maps does not work

### Describe the bug

Spring users can bind maps like this:

```
@ConfigurationProperties("maps")
public class MapWiringProperties {

    // Cover injection of maps of integers and strings;
    public Map<Integer, String> integers;

    // Cover injection of maps of string and classes
    public Map<String, Person> persons;

    public static class Person {
        public String name;
        public int age;

        @Override
        public String toString() {
            return String.format("person[%s:%s]", name, age);
        }
    }
}
```

With application.properties:

```
maps.integers.1=Value 1
maps.integers.2=Value 2

maps.persons.character1.name=Sarah
maps.persons.character1.age=19
maps.persons.character2.name=Terminator
maps.persons.character2.age=999
```

If we try to use the same using the spring properties Quarkus extension, it throws the following exception:

```
Caused by: java.lang.IllegalArgumentException: SRCFG00013: No Converter registered for interface java.util.Map
```

### Expected behavior

Not sure if this is very complex to mimic in the Quarkus extension (we should generate the equivalent object using the `@ConfigMapping` annotation.
However, if this is too complex to support, we should document the current limitations in the Spring properties guide.

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/configproperties/ConfigPropertiesUtil.java
+++ b/extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/configproperties/ConfigPropertiesUtil.java
@@ -4,6 +4,8 @@ import java.util.Collection;
 import java.util.Optional;
 import java.util.function.IntFunction;
 
+import javax.enterprise.inject.spi.DeploymentException;
+
 import org.eclipse.microprofile.config.Config;
 import org.jboss.jandex.DotName;
 import org.jboss.jandex.ParameterizedType;
@@ -42,6 +44,10 @@ final class ConfigPropertiesUtil {
             DotName declaringClass,
             BytecodeCreator bytecodeCreator, ResultHandle config) {
 
+        if (isMap(resultType)) {
+            throw new DeploymentException(
+                    "Using a Map is not supported for classes annotated with '@ConfigProperties'. Consider using https://quarkus.io/guides/config-mappings instead.");
+        }
         if (isCollection(resultType)) {
             ResultHandle smallryeConfig = bytecodeCreator.checkCast(config, SmallRyeConfig.class);
 
@@ -79,6 +85,10 @@ final class ConfigPropertiesUtil {
             BytecodeCreator bytecodeCreator, ResultHandle config) {
 
         ResultHandle optionalValue;
+        if (isMap(resultType)) {
+            throw new DeploymentException(
+                    "Using a Map is not supported for classes annotated with '@ConfigProperties'. Consider using https://quarkus.io/guides/config-mappings instead.");
+        }
         if (isCollection(resultType)) {
             ResultHandle smallryeConfig = bytecodeCreator.checkCast(config, SmallRyeConfig.class);
 
@@ -128,6 +138,11 @@ final class ConfigPropertiesUtil {
         return DotNames.COLLECTION.equals(resultType.name()) ||
                 DotNames.LIST.equals(resultType.name()) ||
                 DotNames.SET.equals(resultType.name());
+    }
+
+    private static boolean isMap(final Type resultType) {
+        return DotNames.MAP.equals(resultType.name()) ||
+                DotNames.HASH_MAP.equals(resultType.name());
     }
 
     static Type determineSingleGenericType(Type type, DotName declaringClass) {

--- a/extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/configproperties/DotNames.java
+++ b/extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/configproperties/DotNames.java
@@ -1,7 +1,9 @@
 package io.quarkus.arc.deployment.configproperties;
 
 import java.util.Collection;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
 import java.util.Optional;
 import java.util.Set;
 
@@ -23,6 +25,8 @@ final class DotNames {
     static final DotName LIST = DotName.createSimple(List.class.getName());
     static final DotName SET = DotName.createSimple(Set.class.getName());
     static final DotName COLLECTION = DotName.createSimple(Collection.class.getName());
+    static final DotName MAP = DotName.createSimple(Map.class.getName());
+    static final DotName HASH_MAP = DotName.createSimple(HashMap.class.getName());
     static final DotName ENUM = DotName.createSimple(Enum.class.getName());
     static final DotName MP_CONFIG_PROPERTIES = DotName
             .createSimple(org.eclipse.microprofile.config.inject.ConfigProperties.class.getName());

--- a/extensions/arc/deployment/src/test/java/io/quarkus/arc/test/configproperties/MapTest.java
+++ b/extensions/arc/deployment/src/test/java/io/quarkus/arc/test/configproperties/MapTest.java
@@ -1,0 +1,41 @@
+package io.quarkus.arc.test.configproperties;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import java.util.Map;
+
+import javax.enterprise.inject.spi.DeploymentException;
+
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.arc.config.ConfigProperties;
+import io.quarkus.test.QuarkusUnitTest;
+
+public class MapTest {
+
+    @RegisterExtension
+    static final QuarkusUnitTest TEST = new QuarkusUnitTest()
+            .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class).addClasses(WithMap.class))
+            .assertException(e -> {
+                assertEquals(DeploymentException.class, e.getClass());
+                assertTrue(e.getMessage().contains("config-mappings"));
+            });
+
+    @Test
+    public void shouldNotBeInvoked() {
+        // This method should not be invoked
+        fail();
+    }
+
+    @ConfigProperties
+    public static class WithMap {
+
+        public String name;
+        public Map<String, String> other;
+    }
+}

